### PR TITLE
Add github actions to export downloadable builds.

### PR DESCRIPTION
**Pull request by fire**
1. Add export presets.
2. Builds Godot Engine stable in Github Actions.
5. Remove the panorama sky for the menu due to build failure. Cannot open the .hdr in krita to convert to .exr.
6. Exports the project.

Special thanks to @RevoluPowered 

https://nightly.link/fire/tps-demo/workflows/builds/builds

Spent a week on this.

## Option to generate pdbs

I can also generate pdbs, but it's optional. Can also strip the executables.

## What is V-Sekai?

We're developing a free, open source virtual reality platform. We work to make it user-friendly and fulfill a need in the VR community. Join the V-Sekai team on the [V-Sekai discord server](https://discord.gg/7BQDHesck8)!

### CODE REVIEW

File: .github/workflows/builds.yml [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+      - uses: actions/checkout@v3
+
+      - uses: actions/checkout@v3
+        with:
+          repository: clayjohn/tps-demo
+          path: tps-demo
+          ref: menu
+
+      - name: Move to the root
+        run: |
```
Comment (fire) [inline .github/workflows/builds.yml:94]:
```suggestion
```

Remove unused section.

File: .github/workflows/static_checks.yml [hunk 2]
```diff
@@ -1,19 +1,77 @@
...
-    name: File formatting (file_format.sh)
+  static-checks:
+    name: Static Checks (clang-format, black format, file format, documentation checks)
     runs-on: ubuntu-20.04
     steps:
-      - name: Checkout
-        uses: actions/checkout@v3
+      - uses: actions/checkout@v3
+        with:
+          repository: v-sekai/godot
```
Comment (fire) [inline .github/workflows/static_checks.yml:15]:
```suggestion
          repository: godotengine/godot
```

Comment (fire) [reply .github/workflows/static_checks.yml:15]: Also lock to the 4.0-stable tag

File: project.godot [hunk 3]
```diff
@@ -13,7 +13,7 @@ config_version=5
 config/name="Godot Third-Person Shooter Demo"
 config/description="Godot Third Person Shooter with high quality assets and lighting"
 run/main_scene="res://main/main.tscn"
-config/features=PackedStringArray("4.0")
+config/features=PackedStringArray("4.0", "Double Precision")
```
Comment (fire) [inline project.godot:16]: Remove double precision.

**Comment by fire** [comment]
Since iteration times are poor, will first fill the cache with deploy: false and then try to fix the build.

**Review by aaronfranke — APPROVED** [review]
I am unable to review at this time but I am approving to unblock. I don't have anything against GitHub Actions exporting builds. Maybe @Calinou could review further.

File: .github/workflows/builds.yml [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+
+      - name: Prepare artifacts
+        if: ${{ matrix.deploy && matrix.deploy-platform != 'windows' }}
+        run: |
+          rm -rf tps_demo_${{ matrix.deploy-platform }}
+          export TARGET=tps_demo_${{ matrix.deploy-platform }}
+          mkdir -p $TARGET
+          cp -rf ${{ matrix.deploy-bin }} $TARGET/tps_demo_${{ matrix.deploy-platform }}
+          cp -rf tps_demo_${{ matrix.deploy-platform }}.pck $TARGET/tps_demo_${{ matrix.deploy-platform }}.pck
+        shell: bash
```
Comment (fire) [inline .github/workflows/builds.yml:138]:
```suggestion
          cp -rf tps_demo_${{ matrix.deploy-platform }}.pck $TARGET/tps_demo_${{ matrix.deploy-platform }}.pck
          7z a -r $TARGET.zip $TARGET
        shell: bash
```

**Comment by Calinou** [comment]
I'd prefer using official export templates to reduce export times and have fully optimized binaries (with LTO).

**Comment by fire** [comment]
I prefer being able to test pre-releases like the upcoming 4.0 point release.

I can remove the C# support, but please let me know if you are still interested.

**Comment by fire** [comment]
I'm running the github action separately with some of your generate releases code from fbx2gltf. 

Here's a sample. https://github.com/V-Sekai/V-Sekai.xr-grid

Thanks. I'll close as it seems this isn't wanted and I don't feel like arguing for it.

**Pull request closed without merging**